CMakeLists.txt

```diff
@@ -0,0 +1,121 @@
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+cmake_minimum_required(VERSION 2.6)
+
+project(ESPResSo)
+
+enable_language(CXX)
+
+# Cmake modules/macros are in a subdirectory to keep this file cleaner
+set(CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
+
+######################################################################
+# User input options                                                   
+######################################################################
+if(NOT CMAKE_BUILD_TYPE)
+    set(CMAKE_BUILD_TYPE "Release" CACHE STRING "Choose the type of build, options are: Debug Release RelWithDebInfo MinSizeRel." FORCE)
+endif(NOT CMAKE_BUILD_TYPE)
+
+# On Mac OS X, first look for other packages, then frameworks
+set(CMAKE_FIND_FRAMEWORK LAST)
+
+option(WITH_MPI    "Build a mpi parallel version of ESPResSo" ON)
+option(WITH_TK     "Build with tk support"  OFF)
+option(WITH_PYTHON "Build python interface"  ON)
+option(WITH_TCL    "Build tcl interface"     ON)
+option(WITH_CUDA   "Build with GPU support"  ON)
+
+# choose the name of the config file
+set(MYCONFIG_NAME "myconfig.hpp"
+  CACHE STRING "Default name of the local config file")
+
+######################################################################
+# Libraries
+######################################################################
+#
+
+find_package(MyConfig)
+
+if (WITH_CUDA)
+  find_package(CUDA REQUIRED)
+  list(APPEND LIBRARIES ${CUDA_CUFFT_LIBRARIES})
+  list(APPEND LIBRARIES ${CUDA_LIBRARIES})
+  set(CUDA 1)
+endif(WITH_CUDA)     
+
+find_package(PythonInterp)
+
+if (WITH_PYTHON)
+  find_package(Cython REQUIRED)
+  find_package(PythonLibs REQUIRED)
+  find_package(NumPy REQUIRED)
+endif(WITH_PYTHON)
+
+find_package(FFTW3)
+if (FFTW3_FOUND)
+  include_directories(${FFTW3_INCLUDE_DIR})
+  list(APPEND LIBRARIES ${FFTW3_LIBRARIES})
+  set(FFTW 3)
+else(FFTW3_FOUND)
+  message(STATUS "FFTW3 not found, diabling it.")
+endif(FFTW3_FOUND)
+
+if (WITH_TCL)
+  find_package(TCL REQUIRED)
+  include_directories(${TCL_INCLUDE_PATH})
+  list(APPEND LIBRARIES ${TCL_LIBRARY})
+endif()
+
+find_package(HDF5)
+if(HDF5_FOUND)
+  set(H5MD 1)
+  list(APPEND LIBRARIES ${HDF5_LIBRARIES})
+  include_directories(${HDF5_INCLUDE_DIRES})
+endif()
+
+#######################################################################
+# Process MPI settings
+#######################################################################
+if(WITH_MPI)
+  find_package(MPI REQUIRED)
+  list(APPEND CMAKE_C_FLAGS ${MPI_COMPILE_FLAGS})
+  list(APPEND CMAKE_EXE_LINKER_FLAGS ${MPI_LINK_FLAGS})
+  include_directories(${MPI_INCLUDE_PATH})
+  list(APPEND LIBRARIES ${MPI_LIBRARIES})
+  set(MPI_SOURCE)
+else(WITH_MPI)
+  include_directories(${CMAKE_SOURCE_DIR}/src/mpifake)  
```

Comment:
Don't need `${CMAKE_SOURCE_DIR}` here, `include_directories` takes paths relative to the source directory.